### PR TITLE
Add b2 and dependencies

### PR DESCRIPTION
**Pull request by freundTech**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by freundTech** [comment]
@conda-forge/help-python ready for review.

File: recipes/b2/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+{% set name = "b2" %}
```
Comment (isuruf) [inline recipes/b2/meta.yaml:1]: This one's too generic of a name and all other package managers (except pip) use it for the b2 build system. Even their git repo is not named `b2` o avoid confusion.

**Comment by isuruf** [comment]
@freundTech I reverted this PR after talking with @ocefpaf. Sorry about that. Can you rename `b2` to something like `backblaze-b2` ? Let me know if you need help.

**Comment by freundTech** [comment]
I already opened https://github.com/conda-forge/staged-recipes/pull/27967 after your comment but before the revert. Is b2-tools fine as a name or would you prefer b2-backblaze? It's the same name as used in homebrew.

**Comment by isuruf** [comment]
I'd prefer `blackblaze-b2` as it's the one used by most distributions. See https://repology.org/project/backblaze-b2/versions

**Pull request merged**